plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryArrowToPageConverter.java

```diff
@@ -0,0 +1,294 @@
+    {
+        if (type instanceof VarcharType) {
+            byte[] slice = ((VarCharVector) vector).get(index);
+            type.writeSlice(output, wrappedBuffer(slice));
+        }
+        else if (type instanceof VarbinaryType) {
+            byte[] slice = ((VarBinaryVector) vector).get(index);
+            type.writeSlice(output, wrappedBuffer(slice));
+        }
+        else {
+            throw new TrinoException(GENERIC_INTERNAL_ERROR, "Unhandled type for Slice: " + type.getTypeSignature());
+        }
+    }
+
+    private void writeObjectLongDecimal(BlockBuilder output, Type type, FieldVector vector, int index)
+    {
+        if (type instanceof DecimalType decimalType) {
+            verify(!decimalType.isShort(), "The type should be long decimal");
+            BigDecimal decimal = vector.getMinorType() == DECIMAL256 ? ((Decimal256Vector) vector).getObject(index) : ((DecimalVector) vector).getObject(index);
+            type.writeObject(output, Decimals.encodeScaledValue(decimal, decimalType.getScale()));
+        }
+        else {
+            throw new TrinoException(GENERIC_INTERNAL_ERROR, "Unhandled type for Object: " + type.getTypeSignature());
+        }
+    }
+
+    private void writeObjectShortDecimal(BlockBuilder output, DecimalType decimalType, FieldVector vector, int index)
+    {
+        verify(decimalType.isShort(), "The type should be short decimal");
+        BigDecimal decimal = vector.getMinorType() == DECIMAL256 ? ((Decimal256Vector) vector).getObject(index) : ((DecimalVector) vector).getObject(index);
+        decimalType.writeLong(output, encodeShortScaledValue(decimal, decimalType.getScale()));
+    }
+
+    private void writeObjectTimestampWithTimezone(BlockBuilder output, Type type, FieldVector vector, int index)
+    {
+        verify(type.equals(TIMESTAMP_TZ_MICROS));
+        long epochMicros = ((TimeStampVector) vector).get(index);
+        int picosOfMillis = toIntExact(floorMod(epochMicros, MICROSECONDS_PER_MILLISECOND)) * PICOSECONDS_PER_MICROSECOND;
+        type.writeObject(output, fromEpochMillisAndFraction(floorDiv(epochMicros, MICROSECONDS_PER_MILLISECOND), picosOfMillis, UTC_KEY));
+    }
+
+    private void writeBlock(BlockBuilder output, Type type, FieldVector vector, int index)
+    {
+        if (type instanceof ArrayType && vector.getMinorType() == LIST) {
+            writeArrayBlock(output, type, vector, index);
+            return;
+        }
+        if (type instanceof RowType && vector.getMinorType() == STRUCT) {
+            writeMapBlock(output, type, vector, index);
+            return;
+        }
+        throw new TrinoException(GENERIC_INTERNAL_ERROR, "Unhandled type for Block: " + type.getTypeSignature());
+    }
+
+    private void writeArrayBlock(BlockBuilder output, Type type, FieldVector vector, int index)
+    {
+        BlockBuilder block = output.beginBlockEntry();
+        Type arrayType = type.getTypeParameters().get(0);
```

Comment:
`arrayType` → `elementType`? Also, please use `getOnlyElement` if the expected size of type parameters is 0.